raiden/api/python.py

```diff
@@ -414,6 +415,25 @@ def channel_open(
 
         return channel_state.identifier
 
+    def mint_token(
+        self,
+        token_address: typing.TokenAddress,
+        to: typing.Address,
+        value: typing.TokenAmount,
+        method: str,
+    ) -> typing.TransactionHash:
+        """ Try to mint `value` units of the token at `token_address` and assign them to `to`,
+        using the minting method named `method`.
+
+        Raises:
+            TransactionNotSuccessful if the minting fails for any reason.
+        """
+        jsonrpc_client = self.raiden.chain.client
+        token_proxy = token_minting_proxy(jsonrpc_client, token_address)
+        args = [to, value] if method == "mint" else [value, to]
```

Comment:
There should be a set of validations here:
1. `to` should be a valid address
2. `value` should not exceed `UINT256_MAX` and should not be less than 0.
3. `contract_method` should be one of the 3 valid contract methods.

Reply:
In the tests, i can see that you are relying on the contract to reject our transaction in case of an invalid input. However, we should prevent the transaction from ever being sent in case the input was invalid to save gas cost.

Reply:
We are not relying on the contract here. The validation is done with the `MintTokenResource` and `MintTokenSchema` classes.